Do not cancel enqueuing if new enqueue is the same engagement type

rasmustautsglia opened the pull request:
**What was solved?**
Currently, if the visitor is enqueuing and decides to initiate engagement again, then that will cancel their queue ticket and create a new one. This is really bad from a UX perspective because a person would potentially lose a great spot in a long queue, even though their reason was to just resume the existing one.

The new solution would check for the engagement kind and if the current one matches the new one, then instead of canceling the old, the old one is resumed.

There were technical difficulties to achieve that because currently, the SDK converts EngagementKind to simplified mediaType, which makes this upgrade impossible. This PR improves the approach significantly and allows the SDK to rely on EngagementKind as long as possible and only use mediaType when absolutely needed. This PR lays a pathway forward to continue removing mediaType from places where EngagementKind is sufficient.

MOB-3931
**Release notes:**

 - [ ] Feature
 - [X] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

- [ ] Is the feature sufficiently tested? All tests fixed? Necessary unit, acceptance, snapshots added? Check that at least new public classes & methods are covered with unit tests
 - [ ] Did you add logging beneficial for troubleshooting of customer issues?
 - [ ] **Did you add new logging?** We would like the logging between platforms to be similar. Refer to **Logging from iOS SDKs** → **Things to consider for newly added logs** in Confluence for more information.

**Screenshots:**

ykyivskyi-gl reviewed (approved):
LGTM. Just a minor comment

rasmustautsglia commented on GliaWidgets/Public/Glia/Glia+EngagementSetup.swift:
```diff
@@ -61,6 +61,15 @@ extension Glia {
         viewFactory: ViewFactory,
         ongoingEngagementMediaStreams: Engagement.Media?
     ) throws {
+        switch interactor.state {
```
fixed [a7d2822](https://github.com/salemove/ios-sdk-widgets/pull/1179/commits/a7d2822d6ec230e0e63a09a5cf57ca3daf1e9cc0)

igorkravchenko commented on GliaWidgets/Public/Glia/Glia+EngagementSetup.swift:
```diff
@@ -61,6 +61,11 @@ extension Glia {
         viewFactory: ViewFactory,
         ongoingEngagementMediaStreams: Engagement.Media?
     ) throws {
+        if case let .enqueued(_, enqueudEngagementKind) = interactor.state, enqueudEngagementKind == engagementKind, let rootCoordinator {
```
I think comment here will be helpful to understand the reason why we need to early out is such case.
Ideally to add uni test here as well (if possible).

igorkravchenko commented on GliaWidgets/Public/Glia/Glia.swift:
```diff
@@ -37,6 +37,21 @@ extension EngagementKind {
         }
     }
+
+    var mediaType: CoreSdkClient.MediaType {
```
It seems like the logic for enqueueing is no longer dependant on `MediaType`, but on `EngagementKind`?
Maybe this mapping can encapsulated in `CoreSdkClient.MediaType` extension, and used (`MediaType(self)`) here? Like so:
Something like:
```swift
extension CoreSdkClient.MediaType {
    init(_ kind: EngagementKind) {
        switch kind {
            ....
        }
    }
}
```
Also it would make sense to add unit test for it.

rasmustautsglia replied:
CoreSDKClient.MediaType extended [18d25dc](https://github.com/salemove/ios-sdk-widgets/pull/1179/commits/18d25dcc689f4b46be4c94c785f8db76c0a1ce7b)

igorkravchenko reviewed (commented):
Looks good. Left two small suggestions.

rasmustautsglia commented:
Refactored GliaViewContorller for testing purposes and removed redundant code. Added test [a58f210](https://github.com/salemove/ios-sdk-widgets/pull/1179/commits/a58f210a49bcabd4b311b247f6d88d2bb1c645c7)

rasmustautsglia commented:
!squash
